api/client/client.go

```diff
@@ -0,0 +1,230 @@
+	if err != nil {
+		return nil, err
+	}
+	client := &Client{
+		client: pb.NewAPIClient(conn),
+		ctx:    context.Background(),
+	}
+	return client, nil
+}
+
+// NewStore cereates a new Store
+func (c *Client) NewStore() (string, error) {
+	resp, err := c.client.NewStore(c.ctx, &pb.NewStoreRequest{})
+	if err != nil {
+		return "", err
+	}
+	return resp.GetID(), nil
+}
+
+// RegisterSchema registers a new model shecma
+func (c *Client) RegisterSchema(storeID, name, schema string) error {
+	req := &pb.RegisterSchemaRequest{
+		StoreID: storeID,
+		Name:    name,
+		Schema:  schema,
+	}
+	_, err := c.client.RegisterSchema(c.ctx, req)
+	return err
+}
+
+// Start starts the specified Store
+func (c *Client) Start(storeID string) error {
+	_, err := c.client.Start(c.ctx, &pb.StartRequest{StoreID: storeID})
+	return err
+}
+
+// StartFromAddress starts the specified Store with the provided address and keys
+func (c *Client) StartFromAddress(storeID string, addr ma.Multiaddr, followKey, readKey *symmetric.Key) error {
+	req := &pb.StartFromAddressRequest{
+		StoreID:   storeID,
+		Address:   addr.String(),
+		FollowKey: followKey.Bytes(),
+		ReadKey:   readKey.Bytes(),
+	}
+	_, err := c.client.StartFromAddress(c.ctx, req)
+	return err
+}
+
+// ModelCreate creates new instances of model objects
+func (c *Client) ModelCreate(storeID, modelName string, items ...interface{}) error {
+	values, err := marshalItems(items)
+	if err != nil {
+		return err
+	}
+
+	req := &pb.ModelCreateRequest{
+		StoreID:   storeID,
+		ModelName: modelName,
+		Values:    values,
+	}
+
+	resp, err := c.client.ModelCreate(c.ctx, req)
+	if err != nil {
+		return err
+	}
+
+	for i, entity := range resp.GetEntities() {
+		err := json.Unmarshal([]byte(entity), items[i])
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// ModelSave saves existing instances
+func (c *Client) ModelSave(storeID, modelName string, items ...interface{}) error {
+	values, err := marshalItems(items)
+	if err != nil {
+		return err
+	}
+
+	req := &pb.ModelSaveRequest{
+		StoreID:   storeID,
+		ModelName: modelName,
+		Values:    values,
+	}
+	_, err = c.client.ModelSave(c.ctx, req)
+	return err
+}
+
+// ModelDelete deletes data
+func (c *Client) ModelDelete(storeID, modelName string, entityIDs ...string) error {
+	req := &pb.ModelDeleteRequest{
+		StoreID:   storeID,
+		ModelName: modelName,
+		EntityIDs: entityIDs,
+	}
+	_, err := c.client.ModelDelete(c.ctx, req)
+	return err
+}
+
+// ModelHas checks if the specified entities exist
+func (c *Client) ModelHas(storeID, modelName string, entityIDs ...string) (bool, error) {
+	req := &pb.ModelHasRequest{
+		StoreID:   storeID,
+		ModelName: modelName,
+		EntityIDs: entityIDs,
+	}
+	resp, err := c.client.ModelHas(c.ctx, req)
+	if err != nil {
+		return false, err
+	}
+	return resp.GetExists(), nil
+}
+
+// ModelFind finds records by query
+func (c *Client) ModelFind(storeID, modelName string, query es.JSONQuery) ([][]byte, error) {
```

Comment:
Would be cool to not have to return `[][]byte`, but not clear to me if that is possible without generics.

Reply:
This method is all for your best interests, so I can change it to `[][]bytes` if you prefer. In fact, it would be even better since returing `[]string` will allocate-extra memory.
If I have your confirmation, I can change that.

Reply:
I'm not totally following what you're saying here, but it makes me think of two things...

1. My original comment was meant to convey that it would be nice for the user to get go-typed data instead of json (`[]byte` or `string`)
2. I think maybe you're talking about making json mode return `[]byte` instead of `string`s. I've actually been thinking that could make more sense. I need to think through the repercussions in other languages, but for Go, it seems to make sense.

Reply:
+1 for `[]byte` return types in JSON mode.

Reply:
Think we need to talk through this in a bit more. I think it's easy to cross wires in this conversation. There are two issues:

1. Should the back end event store code return `[]byte` or `string` for json mode entities? (right now it returns `string` data)
2. Should the client apis return `[]byte` or `string` to the user? (seems that we're setting on `[]byte`, but sometimes we're having to convert from `string` to `[]byte` because of the current state of things related to #1)

I'm not positivities

Reply:
Right, the +1 was for your number 1. 

You number 2 relates to my other comment about settling on one approach. Not so much that one is better than the other, but that we should be consistent. Now I'm thinking, because it's quite nice to pass `interface{}` types into model create, maybe we should also return pre-marshalled `interface{}`s here too?

Reply:
In any case, a chat sounds good! Morning may be best as I'm heading out now for a bit.

Reply:
Yea, let's chat in the morning for sure.

Reply:
issue to track json mode using []byte https://github.com/textileio/go-textile-threads/issues/142

Reply:
Issue to track return type behavior https://github.com/textileio/go-textile-threads/issues/143